### DISCUSSION TITLE
How to encode string to base 64

### DISCUSSION THREAD

**Discussion by gaurikalp** [EnsembleUI/ensemble#89] [Q&A] [answered]
Could you kindly provide guidance on how to encode a string in base 64 in execute code? I want to give the base64-encoded value of the password input to the Login API.
I am using window.btoa but it gives an error

e.g
label: Log In
validateForm: true
onTap:
   action: executeCode
    body: |
        //@code
        var encodedString =  window.btoa(txtPwd.value);

**Comment by kmahmood74** [comment]
hi @gaurikalp I will add that and also share an example here.

**Comment by kmahmood74** [comment]
@gaurikalp just added support for both atob and btoa methods. Examples - 

        var str = 'the ensemble is GREAT';
        var base64 = str.btoa(); //dGhlIGVuc2VtYmxlIGlzIEdSRUFU 
        var equalsEncoded = base64 == btoa(str); //true
        var decoded = atob(base64); //'the ensemble is GREAT'
        var equalsDecoded = decoded == base64.atob(); //true

As you can see, you can call these methods directly on the string or as global functions. Both ways should work.

**Reply by gaurikalp** [reply]
Yes, Its working now. Thank you

**Comment by kmahmood74** [comment]
we'll be uploading the new build shortly, will let you know

**Comment by kmahmood74** [comment, marked as answer]
new build is up on studio.ensembleui.com please check
